pkg/predicate/predicate.go

```diff
@@ -202,3 +205,114 @@ func AddTypePredicate(extensionType string, predicates []predicate.Predicate) []
 	preds = append(preds, HasType(extensionType))
 	return append(preds, predicates...)
 }
+
+// MachineStatusHasChanged is a predicate deciding wether the status of a MCM's Machine has been changed.
+func MachineStatusHasChanged() predicate.Predicate {
+	statusHasChanged := func(oldObj runtime.Object, newObj runtime.Object) bool {
+		oldMachine, ok := oldObj.(*machinesv1alpha1.Machine)
+		if !ok {
+			return false
+		}
+		newMachine, ok := newObj.(*machinesv1alpha1.Machine)
+		if !ok {
+			return false
+		}
+		oldStatus := oldMachine.Status
+		newStatus := newMachine.Status
+
+		//Check the Node
+		if oldStatus.Node != newStatus.Node {
+			return true
+		}
+
+		//Check the CurrentStatus
+		if !equality.Semantic.DeepEqual(oldStatus.CurrentStatus, newStatus.CurrentStatus) {
+			return true
+		}
+
+		if !equality.Semantic.DeepEqual(oldStatus.LastOperation, newStatus.LastOperation) {
+			return true
+		}
+
+		//Check the Conditions
+		if !equality.Semantic.DeepEqual(oldStatus.Conditions, newStatus.Conditions) {
+			return true
+		}
+
```

Comment:
Do we really need to react when the conditions change? What data in the machine status is dynamic that we have to keep for the migration? Isn't it only the `node` field?

Reply:
You are right, it is only the node which we have to keep.
What about the correctness of the Machines and MachineSet states.
Aren't they important for other components or we just leave this to the reconciliation done by MCM?
The predicates will be unusable by future functionalities which are going to depend of this changes in the state.
Do I have to react on changes int the MachineSet at all, then?

Reply:
Hm, good point, especially the `conditions` might trigger remediation if they are `False` for a too long time (and during migration the timestamp might be quite old). So, yes, if you "simply" apply the whole status again after migration to the new `Machine` object then it's probably safer to at least react on `condition` changes. 
Another option could be to only store `.status.node` (also motivated by the fact of reducing the stored data volume) because we know that this is the only field that is needed for the new MCM to adopt the existing node (@hardikdr @prashanth26 @ggaurav10 is this assumption correct)?

For `MachineSet`s, you are probably right and it does not make sense to store any status here because it's not needed for the adoption of the new MCM (again correct @hardikdr @prashanth26 @ggaurav10)?

Reply:
Sorry for the delay, I missed this thread somehow.
Regarding the status of the Machine object, @plkokanov [if I dint forget] did few experiments and it worked fine keeping the only `status.node`,  MCM could reconcile the rest of the status-information. @plkokanov can you please confirm?

Regarding Machineset status, I guess it can also be safely reconciled and we don't need to copy it while migration. Although it would be nice if it could be tested once, I haven't explicitly tested this. 
cc @prashanth26

Reply:
Yep, I experimented with keeping only the status.node field and no status at all for the machinesets. However I only did it with one node and after everything was working (although I doubt that it would be any different otherwise)